cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -144,6 +196,71 @@ protected void runOneIteration() throws Exception {
   }
 
   /**
+   * Perform aggregation on the Streams described by the {@code streamNames}, and no other Streams.
+   * If aggregation was previously done on other Streams, those must be cancelled.
+   *
+   * @param streamNames names of the streams to perform data sizes aggregation on
+   */
+  public void aggregate(Set<String> streamNames) {
+    Set<String> existingAggregators = Sets.newHashSet(aggregators.keySet());
+    for (String streamName : streamNames) {
+      if (existingAggregators.remove(streamName)) {
+        continue;
+      }
+
+      long filesSize = 0;
+      try {
+        StreamConfig config = streamAdmin.getConfig(streamName);
+        filesSize = StreamUtils.fetchStreamFilesSize(config);
+      } catch (IOException e) {
+        LOG.error("Could not compute sizes of files for stream {}", streamName);
+      }
+
+      Aggregator aggregator = new Aggregator(streamName, filesSize);
+      aggregator.startAndWait();
+      aggregators.put(streamName, aggregator);
+    }
+
+    // Stop aggregating the heartbeats we used to listen to before the call to that method,
+    // but don't anymore
+    for (String outdatedStream : existingAggregators) {
+      Aggregator aggregator = aggregators.remove(outdatedStream);
+      aggregator.stopAndWait();
+    }
+  }
+
+  /**
+   * Subscribe to the streams heartbeat notification feed.
+   *
+   * @return a {@link Cancellable} to cancel the subscription
+   * @throws NotificationException if the heartbeat feed does not exist
+   */
+  private Cancellable subscribeToHeartbeatsFeed() throws NotificationException {
+    final NotificationFeed heartbeatsFeed = new NotificationFeed.Builder()
+      .setNamespace("default")
```

Comment:
Notifications uses v3 APIs right? Generally, so far, in v3 APIs, we have tried not to hardcode ``DEFAULT_NAMESPACE`` anywhere. The *namespaceId* is always passed, it could be passed as *default* if the component (e.g Streams in this case) has not been namespaced. However, we have not hardcoded it in v3 APIs. I am unsure of the flow here completely, but like you get stream name, can you get its *accountId*/*namespaceId* as well from the client of these APIs?